srcalc/src/main/java/gov/va/med/srcalc/domain/calculation/HistoricalCalculation.java

```diff
@@ -0,0 +1,282 @@
+    /**
+     * Change this when changing the class!
+     */
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * The maximum length of a station number String, e.g. {@link #getUserStation()}.
+     */
+    public static final int STATION_NUMBER_MAX = 10;
+    
+    /**
+     * The maximum length of a user class String. VistA's maximum appears to be 62.
+     */
+    public static final int PROVIDER_TYPE_MAX = 80;
+
+    private int fId;
+    private String fSpecialtyName;
+    private String fUserStation;
+    private Date fStartTimestamp;
+    private int fSecondsToFirstRun;
+    private Optional<String> fProviderType;
+    
+    /**
+     * Intended for reflection-based construction only. Business code should use the other
+     * constructor.
+     */
+    HistoricalCalculation()
+    {
+    }
+    
+    /**
+     * Constructs an instance with the given properties.
+     * @param specialtyName see {@link #getSpecialtyName()}
+     * @param providerStation see {@link #getUserStation()}
+     * @param secondsToFirstRun see {@link #getSecondsToFirstRun()}
+     * @param providerType see {@link #getProviderType()}.
+     */
+    public HistoricalCalculation(
+            final String specialtyName,
+            final String providerStation,
+            final DateTime startTimestamp,
+            final int secondsToFirstRun,
+            final Optional<String> providerType)
+    {
+        // Use setters to verify constraints.
+        setSpecialtyName(specialtyName);
+        fUserStation = providerStation;
+        // See getStartTimetsamp() for why we enforce 0 millis of second.
+        fStartTimestamp = startTimestamp.withMillisOfSecond(0).toDate();
+        fSecondsToFirstRun = secondsToFirstRun;
+        setProviderType(providerType);
+    }
+    
+    /**
+     * The object's surrogate primary key. Don't show this to the user.
+     */
+    @Id
+    @GeneratedValue
+    public int getId()
+    {
+        return fId;
+    }
+
+    /**
+     * For reflection-based construction only. Business code should never modify
+     * the surrogate key as it is generated from the database.
+     */
+    void setId(final int id)
+    {
+        fId = id;
+    }
+
+    /**
+     * Returns the name of the associated surgical specialty.
+     */
+    @Basic
+    @Column(nullable = false, length = Specialty.SPECIALTY_NAME_MAX)
+    public String getSpecialtyName()
+    {
+        return fSpecialtyName;
+    }
+    
+    /**
+     * For reflection-based construction only. Business code must provide this value to
+     * the constructor.
+     */
+    void setSpecialtyName(final String specialtyName)
+    {
+        fSpecialtyName = Preconditions.requireWithin(
+                specialtyName, 1, Specialty.SPECIALTY_NAME_MAX);
+    }
+
+    /**
+     * Returns the station number of the user who performed the calculation. At time
+     * of writing, this is always the patient's station number as well.
+     * @return never null
+     */
```

Comment:
I think we should investigate if it's possible to have conflicting user/patient station numbers.